pkg/agent/apiserver/handlers/bgproute/handler_test.go

```diff
@@ -161,6 +167,41 @@ func TestBGPRouteQuery(t *testing.T) {
 			url:            "?ipv4-only&ipv6-only",
 			expectedStatus: http.StatusBadRequest,
 		},
+		{
+			name: "get all advertised egressIP routes",
+			url:  "?type=EgressIP",
+			expectedCalls: func(mockBGPServer *queriertest.MockAgentBGPPolicyInfoQuerier) {
+				mockBGPServer.EXPECT().GetBGPRoutes(ctx).Return(allRoutes, nil)
+			},
+			expectedStatus: http.StatusOK,
+			expectedResponse: []apis.BGPRouteResponse{
+				{
+					Route:     egressIPv4Route.Prefix,
+					Type:      string(allRoutes[egressIPv4Route].Type),
+					K8sObjRef: allRoutes[egressIPv4Route].K8sObjRef,
+				},
+				{
+					Route:     egressIPv6Route.Prefix,
+					Type:      string(allRoutes[egressIPv6Route].Type),
+					K8sObjRef: allRoutes[egressIPv6Route].K8sObjRef,
+				},
+			},
+		},
+		{
+			name: "get advertised egress ipv4 routes only",
```

Comment:
actually for consistency with the name of the previous sub-test: "get advertised IPv4 egressIP routes"